File: leetcode/350.java
/*
350. Intersection of Two Arrays II
https://leetcode.com/problems/intersection-of-two-arrays-ii/

Approach: Sorting
*/

import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        
        ArrayList<Integer> arrList = new ArrayList<Integer>();
        
        int i = 0;
        int j = 0;
        while (i < nums1.length && j < nums2.length) {
        	if (nums1[i] == nums2[j]) {
        		arrList.add(nums1[i]);
        		i++;
        		j++;
        	} else if (nums1[i] > nums2[j]) {
        		j++;
        	} else if (nums1[i] < nums2[j]) {
        		i++;
        	}
        }
        
        return convertIntegerArray(arrList);
    }
    
    public int[] convertIntegerArray(ArrayList<Integer> arrList) {
    	int[] intArr = new int[arrList.size()];
    	for (int i = 0; i < intArr.length; i++) {
    		intArr[i] = arrList.get(i);
    	}
    	return intArr;
    }
}

File: leetcode/350.py
'''
350. Intersection of Two Arrays II
https://leetcode.com/problems/intersection-of-two-arrays-ii/

Approach: Sorting
'''


class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1.sort()
        nums2.sort()

        i = 0
        j = 0
        result = []
        while i < len(nums1) and j < len(nums2):
            if nums1[i] == nums2[j]:
                result.append(nums1[i])
                i = i + 1
                j = j + 1
            elif nums1[i] > nums2[j]:
                j = j + 1
            elif nums1[i] < nums2[j]:
                i = i + 1

        return result
